rust/connlib/tunnel/src/peer.rs

```diff
@@ -356,7 +359,7 @@ impl ClientOnGateway {
     // in case that 2 or more resources have overlapping rules.
     fn recalculate_filters(&mut self) {
         self.filters = IpNetworkTable::new();
-        for resource in self.resources.values() {
+        for resource in self.resources.values().filter(|r| r.is_cidr()) {
```

Comment:
Can we delegate to another private function here to make it more obvious that this function is split into several parts, one for each resource type?